tests/kernel/kirk.pm

```diff
@@ -0,0 +1,71 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes kirk testing framework
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+
+use base 'opensusebasetest';
+use testapi qw(get_var get_required_var);
+use utils;
+use strict;
+use testapi;
+use warnings;
+use serial_terminal 'select_serial_terminal';
+
+our $result_file = 'result.json';
+
+sub run
+{
+    my ($self) = @_;
+    my $repo = get_var('KIRK_REPO', 'https://github.com/acerv/kirk.git');
+    my $branch = get_var('KIRK_BRANCH', 'master');
+    my $timeout = get_var('KIRK_TIMEOUT', '5400');
+    my $framework = get_required_var('KIRK_FRAMEWORK');
+    my $sut = get_var('KIRK_SUT', 'host');
+    my $skip = get_var('KIRK_SKIP', '');
+    my $envs = get_var('KIRK_ENVS', '');
+    my $opts = get_var('KIRK_OPTIONS', '');
+    my $suite = get_var('KIRK_SUITE', '');
+
+    select_serial_terminal;
+
+    zypper_call("in -y git");
+    assert_script_run("git clone -q --single-branch -b $branch --depth 1 $repo");
```

Comment:
We already have kirk as package and we should use it, but lets discuss details next week.